### ISSUE TITLE
Support metric and loss kwargs

### ISSUE THREAD

**Issue by cooperlab** [PathologyDataScience/glimr#30]
Support passing of class definitions and kwargs in configurations. 

Currently, we use a `mapper` argument to map strings to metrics and losses since most class instances cannot be passed to trials (non-picklable). This will be replaced with passing class definitions or callables with kwargs.

For metrics - a list of dicts as below (metrics do not contain hyperparameters and can be stored in a list)

`[{“name”: str, “metric”: class/callable, “kwargs”: dict}]`

For losses - a single dict (multiple losses not supported):

`{“name”: str, “loss”: class/callable, “kwargs”: dict}`

**Comment by cooperlab** [comment]
This will impact `keras_losses`, `keras_metrics`, `Search.set_reporter`, `Search.trainable`, and `mnist_example`

https://github.com/PathologyDataScience/glimr/blob/cf1418db786eadf997b8e149d80b15def497d387/glimr/search.py#L221

https://github.com/PathologyDataScience/glimr/blob/cf1418db786eadf997b8e149d80b15def497d387/glimr/search.py#L324C40-L324C51

**Comment by cooperlab** [comment]
Addressed by PR #31
